__init__.py

```diff
@@ -742,37 +741,94 @@ def _fallback_get_alarm_matches(self, utt):
             if user_dow == dt_local.weekday():
                 dow_matches.append(alm)
 
+            if dom == dt_local.day:
+                dom_matches.append(alm)
+
             if have_time:
                 # unambiguous
                 if when_utc == dt_obj:
-                    alarm_matches.append(alm)
+                    exact_matches.append(alm)
             else:
-                self.log.debug("    Loop:have date but no time - disambiguate here!")
-                self.log.debug("    Loop: when:%s, dt_obj:%s" % (when_utc.strftime("%y-%m-%d"), dt_obj.strftime("%y-%m-%d")))
-                # do dates match ?
-                if when_utc.strftime("%y-%m-%d") == dt_obj.strftime("%y-%m-%d"):
-                    self.log.debug("Note, date match with no time")
-                    alarm_matches.append(alm)
+                self.log.debug("    Loop: when:%s, dt_obj:%s" % (when_utc, dt_obj))
+                if when_utc is not None and dt_obj is not None:    
+                    # do dates match ?
+                    if when_utc.strftime("%y-%m-%d") == dt_obj.strftime("%y-%m-%d"):
+                        date_matches.append(alm)
 
-        if len(alarm_matches) == 0:
-            # we can try some heuristics here
-            self.log.info("Entering heuristics phase, dow_matches:%s" % (dow_matches,))
-            alarm_matches = dow_matches
+        if len(exact_matches) > 0:
+            return exact_matches
 
+        if len(date_matches) > 0:
+            return date_matches
 
-        if len(alarm_matches) == 1:
-            return_status = status[1]
-        else:
-            if len(alarm_matches) > 1:
-                self.log.debug("Can't disambiguate, match count %s is > 1"  % (len(alarm_matches,)))
-                self.speak_dialog("alarm.confused")
+        if len(tod_matches) > 0:
+            return tod_matches
 
-        # meet previous ret code protocol
-        if len(alarm_matches) == 0:
-            alarm_matches = None
+        if len(dow_matches) > 0:
+            return dow_matches
 
-        return return_status, alarm_matches
+        return dom_matches
 
+    def get_tod_matches(self, alarm_time, alarm_list):
+        tod_matches = []
+
+        # find alarms where the time matches
+        for alarm in alarm_list:
+            # get utc time from alarm timestamp
+            dt_obj = datetime.fromtimestamp( alarm['timestamp'] )
+            dt_obj = dt_obj.astimezone(pytz.utc)
+
+            # we also need a local version of our utc time
+            cfg_tz = pytz.timezone(self.local_tz) 
+            dt_local = dt_obj.astimezone(cfg_tz)
```

Comment:
Is this the same as `mycroft.util.time.now_local()`?

Reply:
no, this is converting the alarm timestamp in the alarm, not now()